src/vm/amd64/CrtHelpers.asm

```diff
@@ -13,516 +13,264 @@
 ; ***********************************************************************
 
 include AsmMacros.inc
-include asmconstants.inc
 
-; JIT_MemSet/JIT_MemCpy
-;
-; It is IMPORANT that the exception handling code is able to find these guys
-; on the stack, but to keep them from being tailcalled by VC++ we need to turn
-; off optimization and it ends up being a wasteful implementation.
-;
-; Hence these assembly helpers.
-; 
-
-
-;***
-;memset.asm - set a section of memory to all one byte
-;
-; Licensed to the .NET Foundation under one or more agreements.
-; The .NET Foundation licenses this file to you under the MIT license.
-; See the LICENSE file in the project root for more information.;
-;
-;*******************************************************************************
-
-;***
 ;char *memset(dst, value, count) - sets "count" bytes at "dst" to "value"
-;
-;Purpose:
```

Comment:
Can you please retain the header comments that describe the purpose of the method.
Please update the algorithm in the comments to reflect the new strategy.